src/vsphere_cpi/lib/cloud/vsphere/vm_config.rb

```diff
@@ -35,6 +36,18 @@ def name
       @vm_cid
     end
 
+    def calculate_cpu_reservation(cluster)
+      if vm_type.cpu_reserve_full_mhz
+        @maxMhz = 0
+        cluster.host.each do |host|
+          if host.runtime.connection_state == 'connected' &&
+              !host.runtime.in_maintenance_mode
+                @maxMhz = host.summary.hardware.cpu_mhz if host.summary.hardware.cpu_mhz > @maxMhz
+          end
+        end
+      end
+    end
+
```

Comment:
So we're finding the fastest cpu of any host in the cluster and saving that off. Then later, we're multiplying that by the number of CPUs in the VM type to get the actual "cpu_allocation" reservation.

This feels a bit odd. Shouldn't we just do the multiplication here of `host.summary.hardware.cpu_mhz` with `host.summary.hardware. num_cpu_cores` and then use that value later.

Not totally sure it matters in the end though. Is it a requirement that your "vm_type" number of CPUs matches exactly the number of cores in the host to be able to use this feature at all?